internal/provider/function_compare_versions.go

```diff
@@ -64,6 +66,122 @@ func (f CompareVersionsFunction) Run(ctx context.Context, req function.RunReques
 	resp.Error = function.ConcatFuncErrors(resp.Result.Set(ctx, result))
 }
 
+func CompareVersionsRange(version, versionRanges, operator string) (bool, error) {
+	var result bool
+	var err error
+
+	if operator == "inside" || operator == "outside" {
+
+		comparisons, err := GetVersionComparisonsFromVersionRanges(versionRanges)
+		if err != nil {
+			return false, err
+		}
+
+		for _, comparison := range comparisons {
+
+			if comparison.Min == comparison.Max {
+
+				// When the min and max are the same, the version is singular which requires single comparison
+				result, err = CompareVersions(version, comparison.Min, "==")
+				if err != nil {
+					return false, err
+				}
+
+				// If the version is inside the range, the comparison must be true for result to be true
+				// If the version is outside the range, the comparison must be false for result to be true
+				if result {
+					result = true
+					break
+				}
+
+			} else {
+
+				// When the min and max are different, the version is a range
+				// This requires two comparisons, one for each side of the range
+				// Higher or equal than min and lower or equal than max
+				resultMin, err := CompareVersions(version, comparison.Min, ">=")
+				if err != nil {
+					return false, err
+				}
+
+				resultMax := true
+				// If max is "unlimited", we don't need to check the upper bound
+				if comparison.Max != "unlimited" {
+					resultMax, err = CompareVersions(version, comparison.Max, "<=")
+					if err != nil {
+						return false, err
+					}
+				}
+
+				// If the version is inside the range, one comparison must be true for result to be true
+				// If the version is outside the range, all comparison must be true for result to be true
+				if resultMin && resultMax {
+					result = true
+					break
+				}
+			}
+		}
+
+	} else { // If the operator is "==", "!=", ">", "<", ">=", "<=", we assume no range is provided and use the CompareVersions function directly
+
+		result, err = CompareVersions(version, versionRanges, operator)
+		if err != nil {
+			return false, err
+		}
+
+	}
+
+	// If the operator is "outside", invert the result of the comparison that checks inside the range
+	if operator == "outside" {
+		if result {
+			result = false
+		} else {
+			result = true
+		}
+	}
+
+	return result, nil
+}
+
+type VersionComparison struct {
```

Comment:
```suggestion
type VersionRange struct {
```
Or something like that is clearer than Comparison right?